addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -0,0 +1,301 @@
+            'requisition_id': self.id,
+            'notes': self.description,
+            'picking_type_id': self.picking_type_id.id,
+        }
+
+    @api.multi
+    def _prepare_purchase_order_line(self, requisition_line, purchase_id, supplier):
+        self.ensure_one()
+        product = requisition_line.product_id
+        default_uom_po_id = product.uom_po_id.id
+        date_order = self.ordering_date or fields.Datetime.now()
+        qty = self.env['product.uom']._compute_qty(requisition_line.product_uom_id.id, requisition_line.product_qty, default_uom_po_id)
+        taxes = product.supplier_taxes_id
+        fpos = supplier.property_account_position_id.id
+        taxes_id = fpos.map_tax(taxes).ids if fpos else []
+
+        seller = requisition_line.product_id._select_seller(
+            requisition_line.product_id,
+            partner_id=supplier,
+            quantity=qty,
+            date=date_order and date_order[:10],
+            uom_id=product.uom_po_id)
+
+        price_unit = seller.price if seller else 0.0
+        if price_unit and seller and purchase_id.currency_id and seller.currency_id != purchase_id.currency_id:
+            price_unit = seller.currency_id.compute(price_unit, purchase_id.currency_id)
+
+        date_planned = fields.Datetime.to_string(self.env['purchase.order.line']._get_date_planned(seller, po=purchase_id))
+        product_lang = requisition_line.product_id.with_context({
+            'lang': supplier.lang,
+            'partner_id': supplier.id,
+        })
+        name = product_lang.display_name
+        if product_lang.description_purchase:
+            name += '\n' + product_lang.description_purchase
+
+        vals = {
+            'name': name,
+            'order_id': purchase_id.id,
+            'product_qty': qty,
+            'product_id': product.id,
+            'product_uom': default_uom_po_id,
+            'price_unit': price_unit,
+            'date_planned': date_planned,
+            'taxes_id': [(6, 0, taxes_id)],
+            'account_analytic_id': requisition_line.account_analytic_id.id,
+        }
+
+        return vals
+
+    @api.multi
+    def make_purchase_order(self, partner_id):
+        """
+        Create New RFQ for Vendor
+        """
+        assert partner_id, 'Vendor should be specified'
+        supplier = self.env['res.partner'].browse(partner_id)
+        res = {}
+        for requisition in self:
+            if not requisition.multiple_rfq_per_supplier and supplier.id in filter(lambda x: x, [rfq.state != 'cancel' and rfq.partner_id.id or None for rfq in requisition.purchase_ids]):
+                raise UserError(_('You have already one %s purchase order for this partner, you must cancel this purchase order to create a new quotation.') % rfq.state)
+            ctx = dict(self.env.context, mail_create_nolog=True)
+            purchase_id = self.env['purchase.order'].with_context(ctx).create(requisition.with_context(ctx)._prepare_purchase_order(supplier))
```

Comment:
why need to pass context with mail_create_log to call _prepare_purchase_order method ?